src/modules/place/PlaceView.js

```diff
@@ -0,0 +1,266 @@
+import React, {PropTypes, Component} from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  Image,
+  Platform,
+  ActivityIndicator,
+  Dimensions,
+  ScrollView,
+  Linking
+} from 'react-native';
+import * as theme from '../../utils/theme';
+import * as Utils from '../../utils/utils';
+import Button from '../../components/Button';
+
+const window = Dimensions.get('window');
+
+class PlaceView extends Component {
+  static displayName = 'PlaceView';
+
+  static navigationOptions = {
+    title: 'Place',
+    header: {
+      tintColor: theme.colors.text,
+      style: {
+        backgroundColor: theme.colors.navBar
+      }
+    }
+  }
+
+  static propTypes = {
+    office: PropTypes.object.isRequired,
+    loading: PropTypes.bool.isRequired,
+    place: PropTypes.object.isRequired,
+    back: PropTypes.func.isRequired,
+    officeStateActions: PropTypes.shape({
+      retryPlace: PropTypes.func.isRequired
+    }).isRequired
+  };
+
+  componentWillMount() {
+    //if this.prop.place doesn't contain the right data go back to home screen
+    if (!this.props.place.name) {
```

Comment:
Hmm, it seems like the parent of this view should prevent this from happening? Seems a bit hacky.